### PR TITLE
[GUILD] Typescript banner

### PR DESCRIPTION
**Pull request by jelledc**
### Description
This PR converts the `Banner` component to typescript.

### CODE REVIEW

**Review by driesd — DISMISSED** [review]

**Review by lowiebenoot — DISMISSED** [review]

**Review by driesd — DISMISSED** [review]

**Comment by lowiebenoot** [comment]
Waiting for types from Section and Islands, so we can extend on it.

**Comment by jelledc** [comment]
> Waiting for types from [Section](https://github.com/teamleadercrm/ui/pull/2166) and Islands, so we can extend on it.

In the end I decided against this, because that would mean it would also inherit those components prop descriptions. And as these overlap, the `Island` descriptions would always be used, even if it renders as a `Section`.

**Review by driesd — DISMISSED** [review]

**Comment by qubis741** [comment]
Please also export interface, step 4 https://teamleader.atlassian.net/wiki/spaces/ENG/pages/3689152605/Convert+Ahoy+components+to+TypeScript

File: src/components/banner/Banner.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import theme from './theme.css';
+import { IconCloseMediumOutline } from '@teamleader/ui-icons';
+
+interface BannerProps {
+  /** The content to display inside the banner. */
+  children?: React.ReactNode;
+  /** A class name for the wrapper to give custom styles. */
+  className?: string;
+  /** The color of the banner. */
+  color?: 'white' | 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua';
```
Comment (qubis741) [inline src/components/banner/Banner.tsx:16]: use COLORS constant

File: src/components/banner/Banner.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  /** The size of the banner. */
+  size?: 'small' | 'medium' | 'large';
```
Comment (qubis741) [inline src/components/banner/Banner.tsx:18]: use SIZES constant

File: src/components/banner/Banner.tsx (hunk 1 above)
Comment (jelledc) [reply src/components/banner/Banner.tsx]: Changed 👍

File: src/components/banner/Banner.tsx (hunk 2 above)
Comment (jelledc) [reply src/components/banner/Banner.tsx]: Changed 👍

**Review by qubis741 — DISMISSED** [review]

**Review by farazatarodi — DISMISSED** [review]

**Comment by qubis741** [comment]
Manual check failed :rotating_light:, I cant change color

**Review by KristofColpaert — APPROVED** [review]

File: src/components/banner/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+export interface BannerProps extends SectionProps, IslandProps {
+  /** The content to display inside the banner. */
+  children?: React.ReactNode;
+  /** A class name for the wrapper to give custom styles. */
+  className?: string;
+  /** The icon displayed on the left inside the banner. */
+  icon?: React.ReactElement;
+  /** Callback function that is fired when the close button of the banner is clicked. */
+  onClose?(...args: unknown[]): unknown;
```
Comment (lowiebenoot) [inline src/components/banner/Banner.tsx:18]: Q: why is this unknown arguments and unknown return type?

**Review by lowiebenoot — APPROVED** [review]

File: src/components/banner/Banner.tsx (hunk 3 above)
Comment (jelledc) [reply src/components/banner/Banner.tsx:18]: > Q: why is this unknown arguments and unknown return type?

Fixed in follow-up: https://github.com/teamleadercrm/ui/pull/2399

**Pull request merged**